apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,165 @@
+import base64
+import json
+from enum import Enum
+from pathlib import Path
+
+import chevron
+import lxml.html
+import prairielearn as pl
+from typing_extensions import assert_never
+
+
+class Attr(Enum):
+    ANSWER_NAME = "answers-name"
+    WIDTH = "width"
+    HEIGHT = "height"
+    SOURCE_FILE_NAME = "source-file-name"
+    SOURCE_DIRECTORY = "directory"
+
+    @staticmethod
+    def required():
+        return [Attr.ANSWER_NAME.value]
+
+    @staticmethod
+    def optional():
+        return [
+            Attr.WIDTH.value,
+            Attr.HEIGHT.value,
+            Attr.SOURCE_FILE_NAME.value,
+            Attr.SOURCE_DIRECTORY.value,
+        ]
+
+
+class SourceDirectory(Enum):
```

Comment:
You should use this with the `pl.get_enum_attrib` function, as it should simplify the logic here.

Reply:
`get_enum_attrib` forces the values of enums to be lowercase strings but we want the directory to be camelcase because that's the convention other elements use

Reply:
Makes sense, I think you should remove this enum then and replace the `as_runtime_path` function with a dictionary. Then, you can replace the functions here with dictionary containment checks inline, instead of having to define methods for everything.